text/0083-starfish-tracing-model.md

```diff
@@ -0,0 +1,181 @@
+* Have a clear story for indexing and extracing metrics on a per-span level
+* Unified spans and transactions from an SDK perspective
+* Enable a path that allows clients to send 100% of spans outwards at least to a local aggregator,
+  preferrably a remote relay
+* To better and directly support dynamic sampling in the core tracing model
+
+We want to lay out a better path forward that 
+
+* capture entire traces
+* browser tabs = a trace
+* index and extract metrics on a span level
+* clients send 100% of metrics
+* dynamic sampling narrows in on traces
+
+# Terms
+
+The new tracing model is an extension to our existing tracing model.  As such we try go adhere
+to some of the existing terms.  Note that this document is intentionally glossing over some of
+the details to better describe the desired end result.  Individual RFCs will have to be written
+to narrow down on specific schema definitions.
+
+## Session
+
+A session is an optional concept when talking about user actors on the system.  A
+session outlives one or more traces and is exclusively used when talking about human
+interactions with a system.
+
+## Trace
+
+A trace has no end.  It bundles spans together, some of those spans are organized
+into segments (marked in red in the graph).  The user experience does not center
+around a trace, which really is an internal way to bundle things together but it
+narrows down on segments within the trace.
+
+```mermaid
+gantt
+    title Example Starfish Trace
+    dateFormat x
+    axisFormat %S.%L
+
+    section Frontend
+    /checkout                                        :crit, 0, 1500ms
+    GET /api/session                                 :150, 170ms
+    POST /api/analytics                              :190, 70ms
+    GET /api/checkout/state                          :200, 500ms
+    GET /api/checkout/cart                           :1100, 140ms
+    <App/>                                           :1300, 180ms
+    POST /api/analytics                              :done, 1450, 70ms
+    GET /assistent/poll                              :done, 1450, 120ms
+    POST /api/analytics                              :done, 1580, 70ms
+
+    section API Service
+    /api/checkout/state                              :crit, 240, 440ms
+    cache.get session#58;[redacted]                  :360, 10ms
+    db.query select from session                     :370, 20ms
+    db.query select from user                        :390, 20ms
+    db.query select from checkout                    :410, 20ms
+    http.request GET http#58;//payments/poll  :450, 210ms
+    thread.spawn refresh-checkout-cache              :done, 670, 220ms
+
+    section Payment Service
+    /poll                                            :crit, 470, 180ms
+    db.query select from payment                     :490, 30ms
+    db.query update payment                          :530, 60ms
+```
+
+## Span
```

Comment:
I noticed a distinct lack of `span.status` (let's goooo).

Would it be prudent to introduce a basic status field though, similar to OpenTelemetry's? This will have 3 values:

- `ok` (successfully finished)
- `error` (error occurred during span)
- `unknown` (default value)

Reply:
FWIW I did not spec out the entire model here but yes, `status` should be on there.